### ISSUE TITLE
SingleTarget not honored when passing parameters

### ISSUE THREAD

**Issue by johnkors** [fsharp/FAKE#1072]
Not sure if this is intentional, but `-st` / `--single-target` is not honored when passing variables.

Runs only the `UpdateAssemblyInfos` target:
```
Fake.exe UpdateAssemblyInfos --single-target 
```

Runs `UpdateAssemblyInfos` w/ dependencies
```
Fake.exe UpdateAssemblyInfos --single-target var=1
```

I experience no difference when using  `-st` vs `--single-target`.

**Comment by johnkors** [comment]
Duplicate of #594

**Comment by inosik** [comment]
Could you try the latter, but with `single-target=true`? Or maybe `-ev var 1` instead of `var=1`?

I assume that this might be a side effect of mixing "old style" arguments (`var=val`) and "new style" (`-ev var val` and `-ef flag`).

**Comment by bentayloruk** [comment]
@johnkors I believe @inosik is correct.  You have a mix of command line style.  Use `-st` and `-ev`.

**Comment by johnkors** [comment]
I could not find this new vs old style inputs documented anywhere, so I'll leave this as issue as a reference for others. The closest thing is this page: http://fsharp.github.io/FAKE/specifictargets.html , which lists both old and new styles I guess, but not what works with what when using `-st` / `--single-target`


**1 - honors -st (single param)**
```
Fake.exe UpdateAssemblyInfos -st -ev var 1
```
**2 - does not honor -st (several params, note: missing `-ev` before each param )**
```
Fake.exe UpdateAssemblyInfos -st -ev var 1 anothervar 2
```
**3 - honors -st (several params)**
```
Fake.exe UpdateAssemblyInfos -st -ev var 1 -ev anothervar 2
```

Thanks for your help!

**Comment by bentayloruk** [comment]
I agree it is confusing, so I raised https://github.com/fsharp/FAKE/issues/1082.  I'd really appreciate thoughts on that issue, about me introducing a warning.  The problems arise from us supporting two forms of command line options and *not giving notice when we fall back on the "old" style*.

I did the original work to extend the command line.  In order not to break everyone, it had to be backwards compatible with existing scripts.  Hence the new and old mix confusion.  I did document it here http://fsharp.github.io/FAKE/commandline.html  I did not mention the old style, so that new people wouldn't use it.  I just mentioned the version number constraint.  In hindsight, I should have made that clearer.

In addition, I documented it in the FAKE.exe under the **Classic FAKE usage** heading:
![screen shot 2016-01-19 at 08 56 31](https://cloud.githubusercontent.com/assets/189409/12414090/10b59d16-be8b-11e5-9c3e-8347c2e75439.png)
